paddle/framework/attribute.h

```diff
@@ -21,20 +21,12 @@ limitations under the License. */
 #include <vector>
 
 #include "paddle/framework/framework.pb.h"
+#include "paddle/framework/type_defs.h"
 #include "paddle/platform/enforce.h"
-#include "paddle/platform/variant.h"
 
 namespace paddle {
 namespace framework {
 
-// The order should be as same as framework.proto
-typedef boost::variant<boost::blank, int, float, std::string, std::vector<int>,
```

Comment:
What is the reason of moving type `Attribute` from `attribute.h` into a new header files `type_defs.h`? This PR claims to resolve a cyclic dependency issue, but there is no description of this issue.